### PR TITLE
ImagesTable: add error status

### PR DESCRIPTION
**Pull request by kingsleyzissou**
Add error information for failed image builds. The error details are only displayed
for failed builds and the error highest up the error chain is displayed in the expandable
section of the image.

### CODE REVIEW

**Comment by kingsleyzissou** [comment]
image

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    if (err.details && err.details.reason) {
+        return err.details.reason;
+    }
+
+    return err.reason;
+};
+
+const FailureIcon = () => (
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:18]: We could also use a PatternFly `<Alert>` here, specifically the inline plain variation:
https://www.patternfly.org/v4/components/alert/

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import { Flex } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+const useGetErrorReason = (err) => {
+    if (!err || !err.reason) {
+        return 'An unknown error occured';
+    }
+
+    if (err.details && err.details.reason) {
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:11]: We could use optional chaining here instead, i.e. `err.details?.reason`, couldn't we? Obviously the short circuit `&&` totally works and makes sense but for consistency in the code base I think optional chaining is generally preferred when possible, at least that's the impression I have from Jacob from my previous code reviews ;).

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Flex } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+const useGetErrorReason = (err) => {
+    if (!err || !err.reason) {
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:7]: Should we use optional chaining here as well? i.e. `!err2?.reason`?

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        return <React.Fragment />;
+    }
+
+    const reason = useGetErrorReason(status.error);
+
+    return (
+        <div className='pf-u-mt-sm'>
+            <strong>Status</strong>
+            <FailureIcon />
+            <p className='pf-u-danger-color-100 pf-u-w-33-on-md'>
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:38]: Using PatternFly's `<Alert>` would also take care of this, although by default the `<Alert>` text isn't colored red (only the header) so we would still need some CSS.

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        <div className="pf-u-mr-sm">
+            <ExclamationCircleIcon className="error" />
+        </div>
+        <strong className='pf-u-danger-color-100'>Image build failed</strong>
+    </Flex>
+);
+
+const ErrorDetails = ({ status }) => {
+    if (!status || status.status !== 'failure') {
+        return <React.Fragment />;
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:29]: Would the short syntax for fragments `</>` work here? We use both throughout the code base but </>, I'm not sure whether we prefer one over the other... The short syntax is a newer feature...

**Review by lucasgarfield — COMMENTED** [review]
Looks great! Just a few comments, hope it isn't too nit-picky :smile:

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 2 above)
Comment (jkozol) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:11]: Yep this works as expected but `err.details?.reason` accomplishes the same with less code.

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 5 above)
Comment (jkozol) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:29]: So both work. We should ideally only use one. I'm more in favor of the short `<> </>`

File: src/Components/ImagesTable/ImageBuildErrorDetails.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            <strong>Status</strong>
+            <FailureIcon />
+            <p className='pf-u-danger-color-100 pf-u-w-33-on-md'>
+                { reason }
+            </p>
+        </div>
+    );
+};
+
+ErrorDetails.propTypes = {
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageBuildErrorDetails.js:45]: This is not a critical comment and I am not questioning it... but just for my own knowledge, why do we need this?

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 5 above)
Comment (kingsleyzissou) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:29]: I prefer `<></>`, I just saw that one of the other components was using `<React.Fragment>` so went with that for consistency. I'll update.

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 6 above)
Comment (kingsleyzissou) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:45]: Think it's required by the linter to validate the type of props being passed to the component. Think it's just for type safety.

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 1 above)
Comment (kingsleyzissou) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:18]: Ahhhh I didn't think of that. I'll update.

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 4 above)
Comment (kingsleyzissou) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:38]: I guess it makes things slightly simpler. Will have a look

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 2 above)
Comment (kingsleyzissou) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js:11]: Haha my javascript is a bit rusty. Will fix

File: src/Components/ImagesTable/ImageBuildErrorDetails.js (hunk 4 above)
Comment (lucasgarfield) [reply src/Components/ImagesTable/ImageBuildErrorDetails.js]: Oh, good point... I agree that using the `<p>` tag is the way to go here, it is true to the mockup and consistent with the UUID alignment.

**Review by lucasgarfield — APPROVED** [review]
Looks good to me!

**Review by jkozol — APPROVED** [review]

**Pull request merged**